### ISSUE TITLE
Sharing Permissions Oddities

### ISSUE THREAD

**Issue by garyluu** [ga4gh/dockstore#1805]
### Expected behaviour
Sharing a variety of different permissions should work

### Actual behaviour
Sharing owner permissions ends up breaking user/workflow

### Steps to reproduce the behaviour 
- create fresh hosted workflow
- give writer permission to another email
- give reader permission to another user 
- give owner permission to another user

Error is encountered, all permissions removed for the other user, can't give any permissions to the other user anymore.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/DOCK-81)
┆Issue Number: DOCK-81

**Comment by denis-yuen** [comment]
One clue, looks like adding owner permissions in this sequence results in 
```
webservice_1     | WARN  [2018-09-11 15:02:13,009] com.google.api.client.googleapis.services.AbstractGoogleClient: Application name is not set. Call Builder#setApplicationName.
webservice_1     | WARN  [2018-09-11 15:02:13,245] org.glassfish.jersey.client.JerseyInvocation: Entity must not be null for http method PUT.
webservice_1     | ERROR [2018-09-11 15:02:13,312] io.dockstore.webservice.permissions.sam.SamPermissionsImpl: Resource dockstore-tool/%23workflow%2Fdockstore.org%2Fdenis-yuen%2Ftest_cwlx not found
webservice_1     | ! io.swagger.sam.client.ApiException: {"statusCode":404,"source":"sam","causes":[],"stackTrace":[],"message":"Resource dockstore-tool/%23workflow%2Fdockstore.org%2Fdenis-yuen%2Ftest_cwl
x not found"}
webservice_1     | ! at io.swagger.sam.client.ApiClient.invokeAPI(ApiClient.java:737)
webservice_1     | ! at io.swagger.sam.client.api.ResourcesApi.addUserToPolicy(ResourcesApi.java:101)
webservice_1     | ! at io.dockstore.webservice.permissions.sam.SamPermissionsImpl.setPermission(SamPermissionsImpl.java:106)
webservice_1     | ! at io.dockstore.webservice.resources.WorkflowResource.addWorkflowPermission(WorkflowResource.java:884)
```

**Comment by denis-yuen** [comment]
More clues

Monitoring things in SAM. The original owner gets this message

```
https://sam.dsde-staging.broadinstitute.org/api/resources/v1/dockstore-tool/%2523workflow%252Fdockstore.org%252Fdenis-yuen%252Ftest_permissions/policies

{
  "statusCode": 404,
  "source": "sam",
  "causes": [],
  "stackTrace": [],
  "message": "Resource dockstore-tool/%23workflow%2Fdockstore.org%2Fdenis-yuen%2Ftest_permissions not found"
}
```

the user that we tried to add owner access to gets
```
https://sam.dsde-staging.broadinstitute.org/api/resources/v1/dockstore-tool/%2523workflow%252Fdockstore.org%252Fdenis-yuen%252Ftest_permissions/policies

{
  "statusCode": 403,
  "source": "sam",
  "causes": [],
  "stackTrace": [],
  "message": "You may not perform any of [READ_POLICIES] on dockstore-tool/%23workflow%2Fdockstore.org%2Fdenis-yuen%2Ftest_permissions"
}
```

So it seems like the adding an additional owner seems to mess up the ownership by the original user in sam but doesn't quite add the new user successfully.

**Comment by denis-yuen** [comment]
Seems fixed with 1.5.0-rc.2

**Comment by garyluu** [comment]
Seems fine to me
